packages/compiler/core/types.ts

```diff
@@ -258,49 +258,46 @@ export interface TemplateParameterType extends BaseType {
   node: TemplateParameterDeclarationNode;
```

Comment:
Definitely start here, the changes in the other files are mostly the same changes over and over again as fallout from the changes here.